lib/proxy/virsh.rb

```
@@ -17,12 +17,12 @@ def find_ip_for_host host
         return e.parent.attributes["ip"]
       end
     end
-    raise Proxy::DNS::Error.new("Cannot find DNS entry for #{host}")
+    raise Proxy::Dns::NotFound.new("Cannot find DNS entry for #{host}")
   rescue Exception => e
-    msg = "DNS virsh provider error: unable to retrive virsh info: #{e}"
+    msg = "DNS virsh provider error: unable to retrieve virsh info: #{e}"
     logger.error msg
     logger.debug xml if defined?(xml)
-    raise Proxy::DNS::Error, msg
+    raise Proxy::Dns::NotFound.new(msg)
```

Not sure I understand the comment. Do you think it should return an error here?